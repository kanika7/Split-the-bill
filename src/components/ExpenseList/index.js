import React from "react";
import PropTypes from "prop-types";
import { PAGE_TITLE, TOTAL_EXPENSE, SERIALNO, WHOPAID, AMOUNTPAID, DESC } from "../../constants";
import "./styles/index.scss";

const ExpenseList = ({ expenses, total, splitExpense, removeItem }) => (
  <div className="expense-list">
    <ul className="list-wrap">
      <li className="list-item headings">
        <span className="col-xs-12 col-lg-2 d-none d-md-block">{SERIALNO}</span>
        <span className="col-xs-12 col-lg-2">{WHOPAID}</span>
        <span className="col-xs-12 col-lg-2">{AMOUNTPAID}</span>
        <span className="col-xs-12 col-lg-2">{DESC}</span>
      </li>
      {expenses.map((exp, index) => (
        <li className="list-item" key={index}>
          <span className="col-xs-12 col-lg-2 d-none d-md-block">{index + 1}</span>
          <span className="col-xs-12 col-lg-2">{exp.name}</span>
          <span className="col-xs-12 col-lg-2">{exp.amount}</span>
          <span className="col-xs-12 col-lg-2">{exp.description}</span>
          <button type="button" className="close" aria-label="Close" onClick={() => removeItem(index)}>
            <span aria-hidden="true">&times;</span>
          </button>
        </li>
      ))}
      <li className="list-item">
        <span className="col-xs-12 col-lg-3">
          {TOTAL_EXPENSE} : {total}
        </span>
      </li>
    </ul>
    <button type="submit" className="btn btn-primary action-wrapper btn-primary" onClick={splitExpense}>
      {PAGE_TITLE}
    </button>
  </div>
);

ExpenseList.propTypes = {
  expenses: PropTypes.instanceOf(Array).isRequired,
  splitExpense: PropTypes.func.isRequired,
  removeItem: PropTypes.func.isRequired,
  total: PropTypes.number.isRequired
};

export default ExpenseList;
