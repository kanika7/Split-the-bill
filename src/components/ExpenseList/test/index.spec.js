import React from "react";
import { shallow } from "enzyme";
import ExpenseList from "../index";

describe("<ExpenseList /> Test Suits", () => {
  let wrapper;
  const props = {
    expenses: [
      {
        username: "test abc",
        amount: "20",
        description: "test dec"
      }
    ],
    splitExpense: jest.fn(),
    removeItem: jest.fn(),
    total: "200"
  };

  function renderShallow() {
    wrapper = shallow(<ExpenseList {...props} />);
  }

  it("renders snapshot", () => {
    renderShallow();
    expect(wrapper).toMatchSnapshot();
  });

  it("handles removeItem prop method on click of remove", () => {
    const spy = jest.spyOn(props, "removeItem");
    renderShallow();
    wrapper.find(".close").simulate("click");
    expect(spy).toBeCalled();
  });
});
