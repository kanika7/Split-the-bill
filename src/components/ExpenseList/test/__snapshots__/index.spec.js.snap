// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ExpenseList /> Test Suits renders snapshot 1`] = `
<div
  className="expense-list"
>
  <ul
    className="list-wrap"
  >
    <li
      className="list-item headings"
    >
      <span
        className="col-xs-12 col-lg-2 d-none d-md-block"
      >
        S. No.
      </span>
      <span
        className="col-xs-12 col-lg-2"
      >
        Who Paid
      </span>
      <span
        className="col-xs-12 col-lg-2"
      >
        Amount Paid
      </span>
      <span
        className="col-xs-12 col-lg-2"
      >
        Description
      </span>
    </li>
    <li
      className="list-item"
      key="0"
    >
      <span
        className="col-xs-12 col-lg-2 d-none d-md-block"
      >
        1
      </span>
      <span
        className="col-xs-12 col-lg-2"
      />
      <span
        className="col-xs-12 col-lg-2"
      >
        20
      </span>
      <span
        className="col-xs-12 col-lg-2"
      >
        test dec
      </span>
      <button
        aria-label="Close"
        className="close"
        onClick={[Function]}
        type="button"
      >
        <span
          aria-hidden="true"
        >
          Ã—
        </span>
      </button>
    </li>
    <li
      className="list-item"
    >
      <span
        className="col-xs-12 col-lg-3"
      >
        Total Expense
         : 
        200
      </span>
    </li>
  </ul>
  <button
    className="btn btn-primary action-wrapper btn-primary"
    onClick={[MockFunction]}
    type="submit"
  >
    Split The Bill
  </button>
</div>
`;
